async-download-rest/src/main/java/org/uniprot/api/async/download/messaging/consumer/streamer/facade/map/UniProtKBToUniRefMapSolrIdResultStreamerFacade.java

```diff
@@ -0,0 +1,29 @@
+package org.uniprot.api.async.download.messaging.consumer.streamer.facade.map;
```

Comment:
Why do we need UniRefMapSolrIdResultStreamerFacade and UniProtKBToUniRefMapSolrIdResultStreamerFacade. We can merge into one class

Reply:
it simplifies the usage of generics.. the child classes only need to deal with one generic type.. same for other comments